### PR TITLE
Update Istio Gateway Schema

### PR DESCRIPTION
**Pull request by Phlak106**
Update to latest Gateway schema.  The only file edited by hand is apis/istio/v1alpha3/gateway_types.go
see: https://github.com/istio/api/blob/master/networking/v1alpha3/gateway.pb.go

Ran update-deps.sh
Modified gateway_types.go
Ran update-codegen.sh

### CODE REVIEW

**Comment by Phlak106** [comment]
@ZhiminXiang @trshafer @scothis as previous commiters to this file

File: apis/istio/v1alpha3/gateway_types.go [hunk 1]
```diff
@@ -299,6 +324,19 @@ const (
 	// upstream using mutual TLS by presenting client certificates for
 	// authentication.
 	TLSModeMutual TLSMode = "MUTUAL"
+
+	TLSModeAutoPassthrough TLSMode = "AUTO_PASSTHROUGH"
+	TLSModeIstioMutual TLSMode = "ISTIO_MUTUAL"
+)
+
+type TLSProtocol int32
```
Comment (mattmoor-sockpuppet) [inline apis/istio/v1alpha3/gateway_types.go]: Remove trailing whitespace:
```suggestion
type TLSProtocol int32
```

**Comment by trshafer** [comment]
It looks like a lot of additional files were added. Is this large diff expected?

**Comment by Phlak106** [comment]
I ran the update-deps script since the gateway schema is from Istio 1.3.  Was that not necessary?

**Comment by trshafer** [comment]
Thanks for the update. You are correct. `./hack/update-deps.sh` is the correct tooling to use.

Since the PR is so large, could you update the description with the steps you took. It's difficult to see the intention of what changed in such a large PR.

**Comment by Phlak106** [comment]
Can you review apis/istio/v1alpha3/gateway_types.go only?

**Comment by vagababov** [comment]
There's a real istio client now. So it might be better to use it directly rather than keep patching this one?

**Comment by Phlak106** [comment]
Can you point to this Istio client, please?

**Comment by vagababov** [comment]
https://github.com/istio/client-go

**Comment by Phlak106** [comment]
Thanks! Yes much better idea to switch to those.

**Comment by Phlak106** [comment]
Have you been able to use the istio client-go package? When I import it I get errors that DeepCopyInto is not defined

**Comment by vagababov** [comment]
I haven't tried myselft, TBH.

**Comment by Phlak106** [comment]
They said on Istio Slack that their client-go won't be ready for at least 2 weeks.

**Comment by xUnholy** [comment]
Is there any update on the status of this PR and when it's likely to be merged?

**Pull request closed without merging**